### PR TITLE
Modified levenberg_marquardt to recalcuate the jacobian only when necess...

### PR DESCRIPTION
**Pull request by macd**
...ary (ie when the current step is accepted).  Added test that exercises this enhancement

### CODE REVIEW

File: test/levenberg_marquardt.jl [hunk 1]
```diff
@@ -8,4 +8,30 @@ end
 initial_x = [100.0, 100.0]
 
 results = Optim.levenberg_marquardt(f_lm, g_lm, initial_x)
-@assert norm(results.minimum - [0.0, 2.0]) < 0.01
\ No newline at end of file
+@assert norm(results.minimum - [0.0, 2.0]) < 0.01
+
+
+function rosenbrock_res(x, r)
+    r[1] = 10. * (x[2] - x[1]^2 )
```
Comment (johnmyleswhite) [inline test/levenberg_marquardt.jl]: Any chance you'd use `10.0` instead? I find this notation hard to read.

File: test/levenberg_marquardt.jl [hunk 2]
```diff
@@ -8,4 +8,30 @@ end
...
+    r[2] = 1. - x[1]
```
Comment (johnmyleswhite) [inline test/levenberg_marquardt.jl]: 1.0

File: src/levenberg_marquardt.jl [hunk 3]
```diff
@@ -27,6 +27,7 @@ function levenberg_marquardt(f::Function, g::Function, x0; tolX=1e-8, tolG=1e-12
 	converged = false
 	x_converged = false
 	g_converged = false
+        need_jacobian = true
```
Comment (johnmyleswhite) [inline src/levenberg_marquardt.jl]: Seems like there are some spacing issues here.

**Comment by johnmyleswhite** [comment]
Small style comments. Otherwise looks good to go.

File: src/levenberg_marquardt.jl (hunk 3 above)
Comment (macd) [inline src/levenberg_marquardt.jl]: You're right.  I'll change the 10.  to 10.0

On the spacing issue, it looks like the tabs vs. no tabs issue.  I
generally work without tabs, but let me try to put them in and see
how it looks.

Don

**Comment by macd** [comment]
Made changes per John's comments and pushed to my fork.  Do I need to do anything else on this PR?

**Comment by macd** [comment]
oops.  Didn't mean to close.

**Comment by johnmyleswhite** [comment]
Thanks!

**Pull request merged**